### ISSUE TITLE
anchor-plugin not inserting an anchor-tag

### ISSUE THREAD

**Issue by nareshbhatia** [draft-js-plugins/draft-js-plugins#1005]
I have added the anchor-plugin to my editor as documented. When I add a link, it appears to get added correctly to the editorState, however the generated html does not have the `<a>` tag. What am I missing? My code is shown below:

    const linkPlugin = createLinkPlugin();
    const inlineToolbarPlugin = createInlineToolbarPlugin({
        structure: [
            BoldButton,
            ItalicButton,
            UnderlineButton,
            CodeButton,
            linkPlugin.LinkButton
        ]
    });
    const { InlineToolbar } = inlineToolbarPlugin;
    const plugins = [inlineToolbarPlugin, linkPlugin];

    export class TextEdit extends React.Component {
        ...

        render() {
            const { classes } = this.props;

            return (
                <div className={classes.editor}>
                    <Editor
                        editorState={this.state.editorState}
                        onChange={this.onChange}
                        onBlur={this.onBlur}
                        plugins={plugins}
                        ref={element => { this.editor = element; }}
                    />
                    <InlineToolbar />
                </div>
            );
        }

        ...
    }

**Comment by dobesv** [comment]
What do you mean by "generated HTML" ?  Is this something you are looking at in the chrome dev tools?

**Comment by nareshbhatia** [comment]
Exactly! When I look at the generated HTML in the chrome dev tools there is no `<a>` tag there. However, if I do the same in anchor-plugin demo, there is an `<a>` tag. So I must be missing something in my code that I listed above that's preventing the `<a>` tag from being generated.

**Comment by dobesv** [comment]
Hmm, well one thing you should change is to create a new plugin instance for each instance of your component.  e.g. move the calls to `createXXXPlugin` into your constructor.

The plugins seem to assume that they have a 1:1 ratio with the actual editors and keep state within the plugin itself.

So in the constructor you might have `this.plugins = [...]` and reference that in `render()`.

I'm not sure if that's the problem, but it should help.  If not, then I'm not sure, I don't see anything else wrong with the example you posted.

**Comment by nareshbhatia** [comment]
@dobesv, as you suggested, I moved the plugin instantiation to the constructor - that did not help. However I figured out the root cause of the problem. I have a componentDidMount() method in my component which disables the decorators. I am not the first to hit this problem. Here's another issue open on this: https://github.com/draft-js-plugins/draft-js-plugins/issues/955.

cc: @bkniffler, @juliankrispel, @nikgraf - can you please help?

**Comment by nareshbhatia** [comment]
I have now tried a couple of workarounds but none of them work. The first was to move the `componentDidMount()` logic to `componentWillReceiveProps()`, but that works only when the props change, not when the editor is first loaded. Then I tried to move the `componentDidMount()` logic to the constructor. However, that didn't work either - initializing the `editorState` here does not show up it in the editor. I guess the editor is not yet ready in the constructor.

TL;DR: without `componentDidMount()` we are dead in the water. This needs to be fixed.

PLEASE HELP!!

**Comment by nareshbhatia** [comment]
I have finally been able to solve this!

TL;DR: I needed to initialize the editorState from props. If I do this in `componentDidMount()`, then the decorators are lost. However if I do this initialization in the constructor (where props are still available), everything works just fine.

**Old Code**
```
constructor(props) {
    super(props);
    ...
    this.state = {
        editorState: EditorState.createEmpty()
    };
}

componentDidMount() {
    const { html } = this.props;
    this.setState({
        editorState: EditorState.createWithContent(stateFromHTML(html))
    });
}
```

**New Code**
```
constructor(props) {
    super(props);
    ...
    const { html } = this.props;
    this.state = {
        editorState: EditorState.createWithContent(stateFromHTML(html))
    };
}
```

**Comment by mmustafau** [comment]
I dealt with the same problem. Maybe someone needs this solution.

Solution is easy. You should use **onChange** method even though ### readOnly

                        `<Editor
                                    readOnly
                                    editorState={this.state.editorState}
                                    onChange={this.onChange}
                                    plugins={plugins}
                                    ref={(element) => {
                                        this.editor = element;
                                    }}
                                />`

**Comment by ynnelson** [comment]
Still having the issue with `ComponentDidMount()` destroying my decorators... any ideas on how I could still use `ComponentDidMount()` without losing my decorators?

**Comment by mmustafau** [comment]
you can try to set **editorstate** in **componentDidUpdate( )**

**Comment by ynnelson** [comment]
More context in  #1511
